mam/examples/ntru-key-exchange.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 IOTA Stiftung
+ * https:github.com/iotaledger/entangled
+ *
+ * MAM is based on an original implementation & specification by apmi.bsu.by
+ * [ITSec Lab]
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#include <stdio.h>
+#include "mam/api/api.h"
+#include "mam/ntru/ntru.h"
+
+int main() {
+  //Generating a keypair for the receiver api
+  mam_prng_t prng;
+  mam_ntru_sk_t ntru;
+  MAM_TRITS_DEF(nonce, 3 * 10);
+  MAM_TRITS_DEF(key, MAM_PRNG_SECRET_KEY_SIZE);
+  nonce = MAM_TRITS_INIT(nonce, 3 * 10);
+  key = MAM_TRITS_INIT(key, MAM_PRNG_SECRET_KEY_SIZE);
+  trits_from_str(key,
+                 "AAABBBCCCAAABBBCCCAAABBBCCC"
+                 "AAABBBCCCAAABBBCCCAAABBBCCC"
+                 "AAABBBCCCAAABBBCCCAAABBBCCC");
+  mam_prng_init(&prng, key);
+  ntru_sk_reset(&ntru);
+  ntru_sk_gen(&ntru, &prng, nonce);
```

Comment:
`nonce` must be initialized with a certain unique value. It can be a descriptive keypair name (`trits_from_str(nonce, "THIS9IS9RECEIVERS9NTRU9TEST9KEYPAIR99..9")`) or just keypair number. This line here `nonce = MAM_TRITS_INIT(nonce, 3 * 10);` just initializes variable `nonce` so internally it points to the buffer allocated here `MAM_TRITS_DEF(nonce, 3 * 10);`, but it does not initialize trits themselves with any value.

Reply:
Im now initializing nonce